core/types/printing.cc

```diff
@@ -53,7 +53,13 @@ string LiteralType::toStringWithTabs(const GlobalState &gs, int tabs) const {
 }
 
 string LiteralType::show(const GlobalState &gs) const {
-    return fmt::format("{}({})", this->underlying(gs).show(gs), showValue(gs));
+    // RBI generator: Users type the value, not `String("value")`.
+    // if (literalKind == LiteralTypeKind::Symbol) {
+    // HACK: Restore T.deprecated_enum :|
+    return fmt::format("T.deprecated_enum([{}])", showValue(gs));
+    //}
+    // return showValue(gs);
+    // return fmt::format("{}({})", this->underlying(gs).show(gs), showValue(gs));
```

Comment:
This seems wrong. In what cases are we printing literal types?

If we're printing them as keys of `Shape` types, we should probably print them as normal Ruby values.

If they're coming up in places outside of shape types, we should either:

1. not print them. Consider this case:


```
# typed: strict

A = "hello"
```

The GlobalState will show `A` as having type `String("hello")` but we don't actually have to add a `T.let` to the generated RBI to record that fact.

2. Widen them to their underlying type, but this might cause bugs

(tl;dr: i think that we should be able to have a heuristic like "does static field actually need `T.let` annotation" + special handling for shape types)

Reply:
> This seems wrong. In what cases are we printing literal types?

Look at the one rbi-gen test. We have examples at Stripe where these types are used as type parameters.

Reply:
It seems this is specifically to support `T.deprecated_enum`? `T.deprecated_enum` is basically treated as `T.untyped` internally. When we serialize the RBI, we should be able to just put `T.untyped` there, and have the codebase type check equivalently under the generated RBI.

Reply:
The downside to emitting an rbi with `T.untyped` instead would be that hover would become substantially less useful.

Reply:
@elliottt That explanation doesn't quite make sense to me, because Sorbet doesn't have an internal notion of `T.deprecated_enum`. In `type_syntax.cc`, when it sees `T.deprecated_enum([:symbol])`, it's going to convert that to the type `Symbol` anyways. So why not just explicitly widen the type to the underlying when serializing the packager RBI file?

I looked at the example in the test, and still can't quite connect how we're going from "there's a `LiteralType` on the Symbol" to "we should wrap this up in `T.deprecated_enum`". It seems like the two concepts are completely independent in the type system.

Reply:
That works for me -- if we're already turning it into the underlying type, then let's just do that in the generated rbis as well.